From 4bcc82f96b4fb54247cf6e42eea55be73a28aa20 Mon Sep 17 00:00:00 2001
From: Arulselvan M <arulselvan.m@intel.com>
Date: Thu, 30 May 2013 16:47:21 +0530
Subject: [PORT FROM MAIN]MediaPlayerService: check buffer before set it's size to zero

BZ: 69444

Sometimes the output is null, then code set the buffer size to zero, but
in some special case, the buffer(which is pass through info pointer) is
null too, so check it before access the pointer.

Change-Id: I39dd94a85922ce38425e0123dd1d1ca5d9798482
Orig-Change-Id: Idfe5a07b374108de5e4ce5c4c8b6baee30fca652
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 media/libmediaplayerservice/MediaPlayerService.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libmediaplayerservice/MediaPlayerService.cpp b/media/libmediaplayerservice/MediaPlayerService.cpp
index d4037cc..91e5273 100644
--- a/media/libmediaplayerservice/MediaPlayerService.cpp
+++ b/media/libmediaplayerservice/MediaPlayerService.cpp
@@ -1659,7 +1659,7 @@ void MediaPlayerService::AudioOutput::CallbackWrapper(
         // no output set, likely because the track was scheduled to be reused
         // by another player, but the format turned out to be incompatible.
         data->unlock();
-        buffer->size = 0;
+        if (buffer != NULL) buffer->size = 0;
         return;
     }
 
-- 
1.7.4.1

